api/src/main/java/com/messagebird/MessageBirdServiceImpl.java

```diff
@@ -119,9 +120,25 @@ public void deleteByID(String request, String id) throws UnauthorizedException,
 
     @Override
     public <R> R requestList(String request, Integer offset, Integer limit, Class<R> clazz) throws UnauthorizedException, GeneralException {
+        return requestList(request, new Paging.SlicedPaging(offset, limit), clazz);
+    }
+
+    @Override
+    public <R> R requestList(String request, Paging paging, Class<R> clazz) throws UnauthorizedException, GeneralException {
         Map<String, Object> map = new LinkedHashMap<String, Object>();
-        if (offset!=null) map.put("offset", String.valueOf(offset));
-        if (limit!=null) map.put("limit", String.valueOf(limit));
+        if (paging != null) {
+            if (paging instanceof Paging.PagedPaging) {
```

Comment:
Honestly, I would instead make multiple requestList methods that take different paging objects instead of trying to take a base class and casting it to a more specific class. Plus there is no behavior in the base class so it is hierarchy without any benefit.

And honestly, given the API that takes just arguments around paging, it might be better to just provide as raw arguments to the methods instead of encapsulating page/pagesize, and offset/limit into objects. And if you consider the Pagination object that is created from parsing responses, and how it doesn't link to these classes at all, I'd remove the Paging.* and instead use just plain arguments.

Reply:
yes you are right but otherwise returnList was not working properly with offset and limit. We saw this problem in examples so we added this paging thing and didn't break old version. So that what should we do then?

Reply:
Then perhaps an in-between approach is needed. I would untangle the hierarchy of the Paging objects and just have two requestList methods that take either a PagedPaging, or an SlicedPaging object.